src/pip/_internal/operations/check.py

```diff
@@ -113,6 +128,24 @@ def check_install_conflicts(to_install: List[InstallRequirement]) -> ConflictDet
     )
 
 
+def check_unsupported(
+    packages: Iterable[BaseDistribution],
+) -> List[BaseDistribution]:
+    unsupported = []
+    tags = get_supported()
+    for p in packages:
+        with suppress(FileNotFoundError):
+            wheel_file = p.read_text("WHEEL")
+            wheel_tags: FrozenSet[Tag] = reduce(
+                frozenset.union,
+                map(parse_tag, Parser().parsestr(wheel_file).get_all("Tag", [])),
+                frozenset(),
+            )
+            if wheel_tags.isdisjoint(tags):
+                unsupported.append(p)
+    return unsupported
```

Comment:
```suggestion
def check_unsupported(
    packages: Iterable[BaseDistribution],
    supported_tags: Collection[Tag],
) -> Iterator[BaseDistribution]:
    for p in packages:
        with suppress(FileNotFoundError):
            wheel_file = p.read_text("WHEEL")
        wheel_tags: FrozenSet[Tag] = reduce(
            frozenset.union,
            map(parse_tag, Parser().parsestr(wheel_file).get_all("Tag", [])),
            frozenset(),
        )
        if wheel_tags.isdisjoint(supported_tags):
            yield p
```

This makes the function more useful and cheaper to call. (You need to also modify the call sites, of course.)

Reply:
its all should be under `with suppress` block, because it would be failed when wheel_file is not initialized